library/private/options_tools.h

```diff
@@ -215,21 +215,42 @@ direction_t parse(const std::string& str)
 {
   try
   {
-    const std::regex re("([-+]?)([XYZ])", std::regex_constants::icase);
+    const std::regex re("([+-]?x)?([+-]?y)?([+-]?z)?", std::regex_constants::icase);
     std::smatch match;
     if (std::regex_match(str, match, re))
     {
-      const double sign = match[1].str() == "-" ? -1.0 : +1.0;
-      const int index = std::toupper(match[2].str()[0]) - 'X';
-      assert(index >= 0 && index < 3);
-
       direction_t dir;
-      dir[index] = sign;
+      int sign = 1;
+      for (size_t i = 0; i < 3; ++i)
+      {
+        const std::string& match_str = match[i + 1].str();
+        if (!match_str.empty())
+        {
+          if (match_str[0] == '-')
+          {
+            sign = -1;
+          }
+          else if (match_str[0] == '+')
+          {
+            sign = +1;
+          }
+          const int index = std::toupper(match_str[match_str.length() - 1]) - 'X';
+          assert(index >= 0 && index < 3);
+          dir[index] = sign;
+        }
+      }
       return dir;
     }
     else
     {
-      return direction_t(options_tools::parse<std::vector<double>>(str));
+      try
+      {
+        return direction_t(options_tools::parse<std::vector<double>>(str));
+      }
+      catch (const options::parsing_exception&)
+      {
+        throw options::parsing_exception("Cannot parse " + str + " into a direction_t");
```

Comment:
is that a bug fix ?

Reply:
just improves the messages by not raising the "could not parse into a double" that may not be relevant.
For example the input `"foobar"` will not match the regex and be tried as a double vector and fail but "could not parse foobar into a double" is not really useful in that case

Reply:
fair.